frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/YarnTaskExecutor.java

```diff
@@ -0,0 +1,123 @@
+    public YarnTaskExecutor(YarnExecutorConfig config, String taskLocation) {
+        this(config, taskLocation, Collections.<String> emptyList());
+    }
+
+    public YarnTaskExecutor(YarnExecutorConfig config, String taskName, String taskLocation) {
+        this(config, taskName, taskLocation, Collections.<String> emptyList());
+    }
+
+    public YarnTaskExecutor(YarnExecutorConfig config, String taskLocation,
+        List<String> additionalArguments) {
+        this(config, getTaskNameFromLocation(taskLocation), taskLocation, additionalArguments);
+    }
+
+    public YarnTaskExecutor(YarnExecutorConfig config, String taskName, String taskLocation,
+        List<String> additionalArguments) {
+        logger = LoggerFactory.getLogger(getClass());
+        this.config = config;
+        this.taskName = taskName;
+        this.additionalArguments = additionalArguments;
+    }
+
+    private static String getTaskNameFromLocation(String taskLocation) {
+        return taskLocation.replaceAll("^.*/([^/]+)(?:\\.js)?$", "$1");
+    }
+
+    public final void execute(String args, Map<String, String> environment) throws TaskRunnerException {
+        final List<String> arguments = getArguments(args);
+        logger.info("Running " + taskToString(taskName, arguments) + " in " + config.getWorkingDirectory());
+
+        try {
+            final int result =
+                new YarnExecutor(config, arguments, environment).executeAndRedirectOutput(logger);
+            if (result != 0) {
+                throw new TaskRunnerException(
+                    taskToString(taskName, arguments) + " failed. (error code " + result + ")");
+            }
+        } catch (ProcessExecutionException e) {
+            throw new TaskRunnerException(taskToString(taskName, arguments) + " failed.", e);
+        }
+    }
+
+    private List<String> getArguments(String args) {
+        List<String> arguments = new ArrayList<>();
+        if (args != null && !args.equals("null") && !args.isEmpty()) {
+            arguments.addAll(Arrays.asList(args.split("\\s+")));
+        }
+
+        for (String argument : additionalArguments) {
+            if (!arguments.contains(argument)) {
+                arguments.add(argument);
+            }
+        }
+        return arguments;
+    }
+
+    private static String taskToString(String taskName, List<String> arguments) {
+        List<String> clonedArguments = new ArrayList<>(arguments);
+        for (int i = 0; i < clonedArguments.size(); i++) {
+            final String s = clonedArguments.get(i);
+            final boolean maskMavenProxyPassword = s.contains("proxy=");
+            if (maskMavenProxyPassword) {
+                final String bestEffortMaskedPassword = maskPassword(s);
+                clonedArguments.set(i, bestEffortMaskedPassword);
+            }
+        }
+        return "'" + taskName + " " + implode(" ", clonedArguments) + "'";
+    }
+
+    private static String maskPassword(String proxyString) {
```

Comment:
I guess this function could be moved into a separate helper class somewhere? I suspect it's used somewhere else too.